mos-platform/geos-cbm/geos_crt.c

```diff
@@ -0,0 +1,1881 @@
+/***********************************************************************
+ * geos_crt.c
+ *
+ * This file contains the implemantation of the "C" definition of
+ * all geos OS calls for use in GEOS applications  with llvm-mos
+ * C compiler.
+ *
+ * Transcribed from geoProgrammer manual by DiSc.
+ *
+ ***********************************************************************/
+
+#include <geos.h>
+
+/***********************************************************************
+ * GEOS Graphic routines
+ ***********************************************************************/
+
+void DrawPoint(uint16_t x, uint8_t y, bool recover, bool background) {
+  __r3 = x;
+  __r11L = y;
+
+  const unsigned char mode = recover ? 255 : 0;
+  if (background) {
+    __attribute__((leaf)) asm volatile(
+      "clc             \n"
+      "jsr __DrawPoint \n" : : "a"(mode) : "x", "y", "p");
+  } else {
+    __attribute__((leaf)) asm volatile(
+      "sec             \n"
+      "jsr __DrawPoint \n" : : "a"(mode) : "x", "y","p");
+  }
+}
+
+bool TestPoint(uint16_t x, uint8_t y) {
+  bool is_pixel_set;
+
+  __r3 = x;
+  __r11L = y;
+  __attribute__((leaf)) asm volatile(
+      "jsr __TestPoint \n"
+      "ldx #0 \n"
+      "bcc pixel_unset \n"
```

Comment:
The use of a named label will cause problems if TestPoint is ever inlined. (This applies generally throughout this file.) Given that the scope is small, it might be better to just use `1f` and `1:`. Otherwise, `%=` can be used to generate a number that is unique for each ASM block in a translation unit; that number can be appended to the label.